quadra/tasks/segmentation.py

```diff
@@ -126,59 +120,48 @@ def export(self) -> None:
         log.info("Exporting model ready for deployment")
         # Get best model!
         if self.trainer.checkpoint_callback is None:
-            raise ValueError("No checkpoint callback found in the trainer")
-        best_model_path = self.trainer.checkpoint_callback.best_model_path  # type: ignore[attr-defined]
-        log.info("Loaded best model from %s", best_model_path)
-
-        module = self.module.load_from_checkpoint(
-            best_model_path,
-            model=self.module.model,
-            loss_fun=None,
-            optimizer=self.module.optimizer,
-            lr_scheduler=self.module.schedulers,
-        )
+            log.warning("No checkpoint callback found in the trainer, exporting the last model weights")
+            module = self.module
+        else:
+            best_model_path = self.trainer.checkpoint_callback.best_model_path  # type: ignore[attr-defined]
+            log.info("Loaded best model from %s", best_model_path)
+
+            module = self.module.load_from_checkpoint(
+                best_model_path,
+                model=self.module.model,
+                loss_fun=None,
+                optimizer=self.module.optimizer,
+                lr_scheduler=self.module.schedulers,
+            )
 
         if "idx_to_class" not in self.config.datamodule:
             log.info("No idx_to_class key")
-            classes = {0: "good", 1: "bad"}
+            idx_to_class = {0: "good", 1: "bad"}  # TODO: Why is this the default value?
         else:
             log.info("idx_to_class is present")
-            classes = self.config.datamodule.idx_to_class
+            idx_to_class = self.config.datamodule.idx_to_class
 
-        if self.export_config is None:
+        if self.config.export is None:
             raise ValueError(
                 "No export type specified. This should not happen, please check if you have set "
                 "the export_type or assign it to a default value."
             )
 
-        input_shapes = self.export_config.input_shapes
-
-        half_precision = self.trainer.precision == 16
+        half_precision = "16" in self.trainer.precision
 
-        for export_type in self.export_config.types:
-            if export_type == "torchscript":
-                out = export_torchscript_model(
-                    model=module.model,
-                    input_shapes=input_shapes,
-                    output_path=self.export_folder,
-                    half_precision=half_precision,
-                )
-
-                if out is None:
-                    log.warning("Skipping torchscript export since the model is not supported")
-                    continue
-
-                self.exported_model_path, input_shapes = out
```

Comment:
Since `exported_model_path` removed with current code the `Analysis evaluation` after the training is completed cannot be done. Basically `quadra experiment=generic/oxford_pet/segmentation/smp.yaml trainer.max_epochs=2` will fail.